modules/caddyhttp/httpredirectlistener.go

```diff
@@ -20,8 +20,45 @@ import (
 	"net"
 	"net/http"
 	"sync"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
 )
 
+func init() {
+	caddy.RegisterModule(HttpRedirectWrapper{})
+}
+
+// HttpRedirectWrapper provides HTTP->HTTPS redirects for
+// connections that come on the TLS port as an HTTP request,
+// by detecting using the first few bytes that it's not a TLS
+// handshake, but instead an HTTP request.
+//
+// This is especially useful when using a non-standard HTTPS port.
+// A user may simply type the address in their browser without the
+// https:// scheme, which would cause the browser to attempt the
+// connection over HTTP, but this would cause a "Client sent an
+// HTTP request to an HTTPS server" error response.
+//
+// This listener wrapper must be placed BEFORE the "tls" listener
+// wrapper, for it to work properly.
+type HttpRedirectWrapper struct{}
```

Comment:
```suggestion
type HTTPRedirectListenerWrapper struct{}
```

I know it's a mouthful but I think it makes more sense this way

Reply:
Done